commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPIHandler.java

```diff
@@ -615,15 +619,17 @@ void register(CommandMetaData<CommandSender> meta, final Argument[] args,
 			argumentsString.add(arg.getNodeName() + ":" + arg.getClass().getSimpleName());
 		}
 		RegisteredCommand registeredCommandInformation = new RegisteredCommand(commandName, argumentsString, shortDescription,
-			fullDescription, usageDescription, aliases, permission);
+			fullDescription, usageDescription, aliases, permission, namespace);
 		registeredCommands.add(registeredCommandInformation);
+		registeredCommandMap.computeIfAbsent(commandName, k -> new ArrayList<>()).add(registeredCommandInformation);
```

Comment:
My understanding is that `registeredCommandMap` now contains everything that `registeredCommands` contains. Is it possible that we can remove `registeredCommands` now by instead iterating over all lists of values in `registeredCommandMap` so we're not storing `[everything]` and `Map<[commands, [everything]]>`

Reply:
Probably. I haven't considered it. I kept `registeredCommands` because it makes `CommandAPI#getRegisteredCommands()` not change at all. I suppose we could be computing the return value there everytime that method is called. Obviously caching is also an option but considering we support dynamically registering commands I am not sure this is a viable option.
Additionally, I only added the `registeredCommandMap` so `CommandAPIBukkit#fixPermissions()` doesn't have to loop over the whole list everytime a permission is "fixed".

Reply:
While figuring out why the permissions tests aren't working I modified the code so that `registeredCommandMap` is no longer needed. I haven't fully gotten the tests working yet though, so I'm not entirely sure that it works. However, I think I may have found some inconsistent behavior with the current solution, but again I need to test that first.